src/main/java/com/conveyal/gtfs/graphql/GTFSGraphQL.java

```diff
@@ -1,41 +1,93 @@
 package com.conveyal.gtfs.graphql;
 
+import com.conveyal.gtfs.graphql.fetchers.JDBCFetcher;
+import graphql.ExecutionInput;
 import graphql.GraphQL;
+import graphql.execution.instrumentation.dataloader.DataLoaderDispatcherInstrumentation;
+import graphql.introspection.IntrospectionQuery;
+import graphql.schema.DataFetchingEnvironment;
+import org.dataloader.DataLoader;
+import org.dataloader.DataLoaderRegistry;
 
 import javax.sql.DataSource;
-import java.sql.Connection;
-import java.sql.SQLException;
-
-/**
- * This provides a GraphQL API around the gtfs-lib JDBC storage.
- * This just makes a Java API with the right schema available, which uses String requests and responses.
- * To make this into a web API you need to wrap it in an HTTP framework / server.
- */
+import java.util.List;
+import java.util.Map;
+
+import static com.conveyal.gtfs.graphql.fetchers.JDBCFetcher.newJdbcDataLoader;
+
 public class GTFSGraphQL {
 
-    private static DataSource dataSource;
+    /**
+     * Execute a graphQL request.  This method creates a new graphQL runner that is able to cache and batch sql queries
+     * during the course of obtaining the data according to the graphQL query.
+     *
+     * @param dataSource The dataSource to use in connecting to a database to make sql queries.
+     * @param query The graphQL query
+     * @param variables The variables to apply to the graphQL query
+     * @return
```

Comment:
The Map<String, Object> return type is very generic, so probably merits a little explanation. I guess this is returning the objects resulting from the GraphQL request. The Javadoc mentions creating the graphQL runner so my first impression was that it would return that runner. Probably worth clarifying that it creates the runner and actually submits it for execution (to an asynchronous mechanism, but blocking?) then returns the results once complete.